pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -273,18 +219,9 @@ func newVSphere(cfg VSphereConfig) (*VSphere, error) {
 		cfg.Global.RoundTripperCount = RoundTripperDefaultCount
 	}
 
-	c, err := newClient(context.TODO(), &cfg)
-	if err != nil {
-		return nil, err
-	}
-
-	id, err := getVMName(c, &cfg)
-	if err != nil {
-		return nil, err
-	}
+	id := cfg.Global.VMName
 
 	vs := VSphere{
-		client:          c,
```

Comment:
If you remove this, you have call vSphereLogin() every time to renew the client where it basically initializes it. I see a function NodeExists() which might get called when the vs.client is not initialized and will eventually run into runtime exception. 

Why don't you want to keep it, as at the very first time the client is initialized and we are in safe hands.